mods/default/mapgen.lua

```diff
@@ -1655,9 +1754,18 @@ end
 
 
 --
--- Detect mapgen to select functions
+-- Detect mapgen, flags and parameters to select functions
 --
 
+-- Get setting or default
+local mgv7_spflags = minetest.get_mapgen_setting("mgv7_spflags") or
+	"mountains, ridges, nofloatlands"
+local captures_float = string.match(mgv7_spflags, "floatlands")
```

Comment:
Yes it may look weird, but there are issues with how this works, i spent a few hours testing and refining the code to work in all cases and this is the result.